paginator/rule.go

```diff
@@ -1,13 +1,25 @@
 package paginator
 
-import "github.com/pilagod/gorm-cursor-paginator/v2/internal/util"
+import (
+	"reflect"
+
+	"github.com/pilagod/gorm-cursor-paginator/v2/internal/util"
+)
 
 // Rule for paginator
 type Rule struct {
 	Key             string
 	Order           Order
 	SQLRepr         string
 	NULLReplacement interface{}
+	CustomType      *CustomType
+}
+
+// CustomType for paginator. It provides extra info needed to paginate across custom types (e.g. JSON)
+type CustomType struct {
+	Meta    interface{}
+	Type    reflect.Type
+	SQLType string
```

Comment:
Would you think it's reasonable to move `SQLType` from `CustomType` to `Rule`? I feel `SQLType` is a general property that can be applied to both basic and custom type.

Reply:
Yes! Let's move it to the `Rule`.